packages/puppeteer-core/src/node/ChromeLauncher.ts

```diff
@@ -43,6 +44,27 @@ export class ChromeLauncher extends ProductLauncher {
     super(puppeteer, 'chrome');
   }
 
+  override launch(options: PuppeteerNodeLaunchOptions = {}): Promise<Browser> {
+    const headless = options.headless ?? true;
+    if (
+      headless === true &&
+      this.puppeteer.configuration.logLevel !== 'silent'
+    ) {
+      console.warn(
+        [
+          '\u001b[1m\u001B[43m\u001b[30m',
+          'Puppeteer `Old headless` deprecation warning:\u001b[0m\u001b[33m',
+          '  In the near feature "headless" for Chrome will run `--headless=new` by default.',
+          '  For more information, please see https://developer.chrome.com/articles/new-headless/.',
+          '  Consider opting in early by passing `headless: "new"` to `puppeteer.launch()`',
+          '  If you encounter any bugs, please report them to https://github.com/puppeteer/puppeteer/issues/new/choose.\u001b[0m\n',
```

Comment:
```suggestion
          '\x1B[1m\x1B[43m\x1B[30m',
          'Puppeteer old Headless deprecation warning:\x1B[0m\x1B[33m',
          '  In the near feature `headless: true` will default to the new Headless mode',
          '  for Chrome instead of the old Headless implementation. For more',
          '  information, please see https://developer.chrome.com/articles/new-headless/.',
          '  Consider opting in early by passing `headless: "new"` to `puppeteer.launch()`',
          '  If you encounter any bugs, please report them to https://github.com/puppeteer/puppeteer/issues/new/choose.\x1B[0m\n',
```